src/components/AddSlots/b.js

```diff
@@ -0,0 +1,215 @@
+
+import Alerts from '../Alerts';
+import { API_URL } from '../../common/consts';
+
+
+export default function AddSlots({
+  toggleAddSlots,
+  user,
+  openModal,
+  ...others
+}) {
+  let maxPeopleAllowedRef = useRef(null);
+  const [allSlots, setAllSlots] = useState([]);
+  const [error, setError] = useState({});
+  const [selectedTime, setSelectedTime] = useState({
+    start_minutes: 0,
+    start_hours: 0,
+    end_hours: 0,
+    end_minutes: 0,
+    maximun_people_allowed: 5
+  });
+
+  useEffect(() => {
+    getSlots();
+  }, []);
+
+  const showError = (type, message) => setError({ type, message });
+  const closeError = () => setError({ type: '', message: '' });
+
+
+  const getSlots = () => {
+    const storeId = user && user.storeId;
+    const filter = { where: { storeId: storeId } };
+    axios
+      .get(
+        `${API_URL}/stores_slots?filter=${JSON.stringify(
+          filter
+        )}`
+      )
+      .then(res => {
+        setAllSlots(res.data);
+      })
+      .catch(err => {
+        showError('danger', 'Error in fetching all slots');
+      });
+  };
+
+  const deleteSlots = slotId => {
+    closeError();
+    axios
+      .delete(
+        `${API_URL}/stores_slots/${slotId}`
+      )
+      .then(res => {
+        getSlots();
+      })
+      .catch(err => {
+        showError('danger', 'Error in deleting slot');
+      });
+  };
+
+  const addSlots = timeString => {
+    closeError();
+    const storeId = user && user.storeId;
+
+    const timeArray = timeString.split(' - ');
+    const startTime = timeArray && timeArray[0].split(':');
+    const endTime = timeArray && timeArray[1].split(':');
+    const startHours = startTime && startTime[0];
+    const startMinutes = startTime && startTime[1];
+    const endHours = endTime && endTime[0];
+    const endMinutes = endTime && endTime[1];
+
+    console.log(timeArray[0]);
+    console.log(startTime, endTime);
+
+    const body = {
+      start_minutes: startMinutes,
+      start_hours: startHours,
+      end_hours: endHours,
+      end_minutes: endMinutes,
+      maximun_people_allowed: document.getElementById(timeString).value || 5,
+      storeId: storeId
+    };
+
+    axios
+      .post(`${API_URL}/stores_slots`, {
+        ...body
+      })
+      .then(res => {
+        getSlots();
+      })
+      .catch(err => {
+        showError('danger', 'Error in adding slot');
+      });
+  };
+
+  const printAllSlots = () => {
+    const store = user.stores || {};
+    const startHour = parseInt(store.shop_open_hours || 0);
+    const endHour = parseInt(store.shop_close_hours || 23);
+    const startMinutes = parseInt(store.shop_open_minutes || 0);
+    const endMinutes = parseInt(store.shop_close_hours || 0);
+    const slotDuration = parseInt(store.slot_duration || 15);
+
+    const buttons = [];
+
+    let count = 0;
+
+    for (let i = startHour; i <= endHour; i++) {
+      let timeString = '';
+      let minutes = 0;
+      while (minutes <= 45) {
```

Comment:
this might not work correctly if a store owner decides to use 10 minute slots, or 5 minute slots. (edge case).